pkg/generate/app/cmd/newapp.go

```diff
@@ -717,13 +717,86 @@ func (c *AppConfig) checkCircularReferences(objects app.Objects) error {
 	for _, obj := range objects {
 		if bc, ok := obj.(*buildapi.BuildConfig); ok {
 			input := buildutil.GetInputReference(bc.Spec.Strategy)
-			if bc.Spec.Output.To != nil && input != nil &&
-				reflect.DeepEqual(input, bc.Spec.Output.To) {
-				ns := input.Namespace
-				if len(ns) == 0 {
-					ns = c.OriginNamespace
+			output := bc.Spec.Output.To
+
+			if output == nil || input == nil {
+				return nil
+			}
+
+			inNS := input.Namespace
+			if len(inNS) == 0 {
+				inNS = c.OriginNamespace
+			}
+
+			// We might be outputing a docker image that would trigger our imagestream repo.
+			if output.Kind == "DockerImage" && input.Kind == "ImageStreamTag" {
+				isName, isTag, ok := imageapi.SplitImageStreamTag(input.Name)
+				if !ok {
+					return fmt.Errorf("Unable to parse ImageStreamTag name: %q", input.Name)
 				}
-				return app.CircularOutputReferenceError{Reference: fmt.Sprintf("%s/%s", ns, input.Name)}
+
+				// The imagestream is usually being created alongside the buildconfig
+				// when new-build is being used, so scan objects being created for it.
+				var is *imageapi.ImageStream
+				for _, check := range objects {
+					if is2, ok := check.(*imageapi.ImageStream); ok {
+						if is2.Name == isName {
+							is = is2
+							break
+						}
+					}
+				}
+
+				if is == nil {
+					var err error
+					is, err = c.OSClient.ImageStreams(inNS).Get(isName)
+					if err != nil {
+						glog.Warningf("Unable to check for circular build input/outputs")
```

Comment:
i'm not sure we want to warn here... it's not uncommon to be defining a build based on an existing imagestream (one of the default s2i imagestreams), or am i misunderstanding the circumstances under which this warning would be printed?